node/src/mempool.rs

```diff
@@ -242,8 +243,28 @@ impl MempoolSrv {
         })?;
 
         // VM Preverify call
-        if let Err(e) = vm.read().await.preverify(tx) {
-            Err(TxAcceptanceError::VerificationFailed(format!("{e:?}")))?;
+        let preverification_data =
+            vm.read().await.preverify(tx).map_err(|e| {
+                TxAcceptanceError::VerificationFailed(format!("{e:?}"))
+            })?;
+
+        if let PreverificationResult::FutureNonce {
+            account,
+            state,
+            nonce_used,
+        } = preverification_data
+        {
+            db.read().await.view(|db| {
+                for nonce in state.nonce + 1..nonce_used {
+                    let spending_id = SpendingId::AccountNonce(account, nonce);
+                    if db.get_txs_by_spendable_ids(&[spending_id]).is_empty() {
```

Comment:
Not needed for now, but in the future it would be interesting to see what an "inverse" implementation would do, where we pre-compute the AccountNonce. This should be limited to a maximum range to prevent spam attacks.

Since we limit the mempool to 10k transactions, we can already set that as a natural limit per account, but logically it would be a lower number.

If the current implementation is performant enough, I would discard this comment completely.

Reply:
As you said, let's bench it